### DISCUSSION TITLE
Getting build error in vercel for getstaticProps pages since last build

### DISCUSSION THREAD

**Discussion by de-shine** [vercel/next.js#49251] [Help]
### Summary

In Vercel the next js project build suddenly starts failing for all getStaticProps pages. any fix ?

### Additional information

```js
Check below error logs:
Error occurred prerendering page "/services/drywall-repair-1". Read more: https://nextjs.org/docs/messages/prerender-error
TypeError: Cannot read properties of null (reading 'useContext')
    at exports.useContext (/vercel/path0/node_modules/react/cjs/react.production.min.js:24:118)
    at $704cf1d3b684cc5c$export$9f8ac96af4b1b2ae (file:///vercel/path0/node_modules/@react-aria/ssr/dist/import.mjs:39:37)
    at Ge (/vercel/path0/node_modules/next/dist/compiled/react-dom/cjs/react-dom-server.browser.production.min.js:114:273)
    at Z (/vercel/path0/node_modules/next/dist/compiled/react-dom/cjs/react-dom-server.browser.production.min.js:120:91)
    at Ge (/vercel/path0/node_modules/next/dist/compiled/react-dom/cjs/react-dom-server.browser.production.min.js:115:9)
    at Z (/vercel/path0/node_modules/next/dist/compiled/react-dom/cjs/react-dom-server.browser.production.min.js:120:91)
    at Ge (/vercel/path0/node_modules/next/dist/compiled/react-dom/cjs/react-dom-server.browser.production.min.js:115:9)
    at Z (/vercel/path0/node_modules/next/dist/compiled/react-dom/cjs/react-dom-server.browser.production.min.js:120:91)
    at Ge (/vercel/path0/node_modules/next/dist/compiled/react-dom/cjs/react-dom-server.browser.production.min.js:115:9)
    at Z (/vercel/path0/node_modules/next/dist/compiled/react-dom/cjs/react-dom-server.browser.production.min.js:120:91)
```

**Comment by laubonghaudoi** [comment]
+1 I am having the same error. Mine is `TypeError: Cannot read properties of null (reading 'useEffect')` instead of `useContext`. I suspect that this is caused by React hooks not accessible from server components. But I am using the `/pages` route not `app/`. Do we need to specify client components in `/pages` too?  I tried adding 'use client' to the top of my component files and the error still persists. It could be a bug introduced in 13.4.

I assumed that upgrading to 13.4 is not breaking for projects that continue to use `pages/`. My local builds are successful and it only fails on Vercel.

Edit: opened a bug at https://github.com/vercel/next.js/issues/49261

**Reply by icyJoseph** [reply]
It should not matter. This must be a bug.

And yeah it looks like:

```jsx
experimental: {
  appDir: false,
},
```

Does the trick, but it shouldn't be necessary.

**Comment by joepetrillo** [comment]
I am having similar issues. My build works locally but Vercel does not like it. I am also only using the pages directory. 

I am getting lots and lots of errors, but here are some which seem to follow a theme.

` TypeError: Cannot read properties of null (reading 'useState')`
` Error occurred prerendering page "/404"`
` TypeError: Cannot read properties of null (reading 'useContext')`

And so on...

**Comment by lucasishuman** [comment]
Same situation here. Even after rolling back to Next 13.3.4, which _had_ been working to me before trying 13.4, and redeploying I'm still seeing issues on Vercel (but not locally) for pages using `getServerSideProps`. No other Next config changes. I'm also only using the `pages` directory.

E.g.
```
TypeError: Cannot read properties of null (reading 'useState')
    at exports.useState (/var/task/node_modules/react/cjs/react.production.min.js:25:394)
...
```

The site builds and deploys OK -  but upon request, it bombs...

![image](https://user-images.githubusercontent.com/1753761/236355764-0dad516b-b100-4d07-8a86-99ab0b487b82.png)

![image](https://user-images.githubusercontent.com/1753761/236356234-324ec280-4dd3-47c4-befe-1ce5f69faf81.png)

**Comment by laubonghaudoi** [comment]
Opened a bug here https://github.com/vercel/next.js/issues/49261

**Comment by skaraudio** [comment]
I concur, we are also facing major issues. Tried reverting, doesn't worth either. Please help us to fix this as soon as possible. Our 50+ person company relies extensively on our backend with NextJs.

<img width="1299" alt="image" src="https://user-images.githubusercontent.com/15099697/236360860-53792e0c-c9b5-4e0e-a877-51eae67ae5f2.png">

**Reply by kgilliam125** [reply]
@skaraudio FYI that you can revert to Nextjs 13.3.4 to get back to a working state. I'm at least seeing success with my projects on this.

**Comment by drsh4dow** [comment]
Same here; we are facing the same problem at my company. 

![image](https://user-images.githubusercontent.com/45247070/236361058-3bb23c86-0ab0-4e21-8c0e-f36afcd1e5e4.png)

Rolling back to Next 13.3.4 solves the problem tho

**Comment by skaraudio** [comment]
Even with 13.3.4 I get this error.. 

<img width="1222" alt="image" src="https://user-images.githubusercontent.com/15099697/236362866-c1af56ab-95fe-4459-bf7e-50a41517762f.png">

**Reply by de-shine** [reply]
+1. Its the same error for my company website as well.

**Comment by w3bdesign** [comment]


**Reply by TravisHi** [reply]
> @de-shine
> 
> Try adding the following to next.config.js:
> 
> ```
> experimental: {
>   appDir: false,
> },
> ```

Thank you! This was the solution for us (along with updating Next to 13.4.0 in our package.json).

**Reply by de-shine** [reply]
Thank you so much. The above config worked for me.

**Comment by jd-carroll** [comment]
Also facing the same issue, here is a simple repro case:

#### Step 1:
```
test % npx create-next-app@latest --typescript
✔ What is your project named? … my-app
✔ Would you like to use ESLint with this project? … Yes
✔ Would you like to use Tailwind CSS with this project? … No
✔ Would you like to use `src/` directory with this project? … Yes
✔ Use App Router (recommended)? … No
✔ Would you like to customize the default import alias? … No
Creating a new Next.js app in /Users/my-profile/develop/test/my-app.

...
```

#### Step 2:
**DO NOT cd INTO THE NEWLY CREATED APP!**

```
test % npm install @vercel/analytics
```

#### Step 3:

```
test % cd my-app
my-app % npm run dev

> my-app@0.1.0 dev
> next dev

ready - started server on 0.0.0.0:3000, url: http://localhost:3000
event - compiled client and server successfully in 951 ms (169 modules)
wait  - compiling / (client and server)...
event - compiled client and server successfully in 626 ms (198 modules)
Warning: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://reactjs.org/link/invalid-hook-call for tips about how to debug and fix this problem.
TypeError: Cannot read properties of null (reading 'useEffect')
    at useEffect (/Users/joseph/develop/andook/andook-client-workbench/node_modules/react/cjs/react.development.js:1634:21)
    at Analytics (file:///Users/joseph/develop/andook/andook-client-workbench/node_modules/@vercel/analytics/dist/react/index.js:132:3)
    at renderWithHooks (/Users/joseph/develop/andook/andook-client-workbench/test/my-app/node_modules/react-dom/cjs/react-dom-server.browser.development.js:5658:16)
    at renderIndeterminateComponent (/Users/joseph/develop/andook/andook-client-workbench/test/my-app/node_modules/react-dom/cjs/react-dom-server.browser.development.js:5731:15)
    at renderElement (/Users/joseph/develop/andook/andook-client-workbench/test/my-app/node_modules/react-dom/cjs/react-dom-server.browser.development.js:5946:7)
    at renderNodeDestructiveImpl (/Users/joseph/develop/andook/andook-client-workbench/test/my-app/node_modules/react-dom/cjs/react-dom-server.browser.development.js:6104:11)
```

**Reply by icyJoseph** [reply]
Why wouldn't you CD into the newly created project?

**Reply by jd-carroll** [reply]
If you cd into the directory, it will work for some reason. Installing in the parent directory (similar to a monorepo setup) will force the error to occur.

**Comment by icyJoseph** [comment]
Alright, this has been addressed here: https://github.com/vercel/next.js/pull/49274

For now explicitly setting appDir to false in the experimental block, works fine.

```js
experimental: {
  appDir: false,
},
```
However, that setting will also be removed in an upcoming version: https://github.com/vercel/next.js/pull/49241#event-9174121381

**Comment by w3bdesign** [comment]
This issue has been resolved in 3.4.1 and you no longer need the `appDir: false` setting in next.config.

I removed it myself and my applications work. 

@de-shine

**Comment by laubonghaudoi** [comment]
Just upgraded to 13.4.2 and this issue is occurring again. See https://github.com/vercel/next.js/issues/49261#issuecomment-1545222421

**Comment by Athrun-Judah** [comment]
the same error in 14.0.2
